internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -974,6 +975,14 @@ func (r *ClusterDeploymentsReconciler) createNewCluster(
 		return ctrl.Result{Requeue: true, RequeueAfter: longerRequeueAfterOnError}, nil
 	}
 
+	platformType := v1beta1.PlatformTypeBaremetal
+	if clusterInstall.Spec.Platform != nil {
+		platformType = clusterInstall.Spec.Platform.PlatformType
+	}
+	if isUserManagedNetwork(clusterInstall) {
```

Comment:
Do you think it would be better to fail validation and ask the user to explicitly set `Platform: None` if the network is user-managed?

I would rather have the user do this explicitly than automatically setting the platform for them.

Reply:
If we change that, it will break compatibility since SNO is `UserManagedNetwork`. It means that CRs for SNO will not work any more without having to set the platform explicitly.